### PR TITLE
fix(duckdb): ensure that parameter names are unlikely to overlap with column names

### LINKED ISSUE #7676: bug: argument name for lambda in Array.map() can't conflict with existing Column names
Prrobably related to https://github.com/ibis-project/ibis/issues/7626

### What happened?

```python
import ibis

ibis.options.interactive = True

t = ibis.memtable({"x": [[1, 2, 3]]})
# t.x.map(lambda y: y + 1)  # works fine
t.x.map(lambda x: x + 1)
```

results in `BinderException: Binder Error: Invalid parameter name 't0.x': must be unqualified`

### What version of ibis are you using?

11e03fa7003ff05bfc243a7cd16b0f791b488737

### What backend(s) are you using, if any?

duckdb

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
Fixes #7676.

I created https://github.com/duckdb/duckdb/issues/9981 because I think this is a bug in DuckDB.

For now this is a workaround to address the issue.

### CODE REVIEW

File: ibis/expr/operations/core.py [hunk 1]
```diff
@@ -183,5 +184,9 @@ class Argument(Value):
     shape: ds.DataShape
     dtype: dt.DataType
 
+    @attribute
+    def param(self) -> str:
```
Comment (kszucs) [inline ibis/expr/operations/core.py:188]: I am confusing it with `ops.ScalarParameter` based in the attribute name. By the way this seems to be an implementation detail of the sql backends (in particular duckdb), could you implement the "unique parameter name" generation in the compiler instead?

Comment (cpcloud) [reply ibis/expr/operations/core.py:188]: I'd rather not.

Comment (cpcloud) [reply ibis/expr/operations/core.py:188]: I have to do it for each backend individually then, and that's not worth the benefit.

Comment (kszucs) [reply ibis/expr/operations/core.py:188]: Okey then, keep it :)

**Review by kszucs — APPROVED** [review]

**Pull request merged**